netpyne/batch.py

```diff
@@ -42,26 +57,184 @@ def tupleToStr (obj):
                 obj[str(key)] = obj.pop(key) 
     #print 'after:', obj
     return obj
+    
+# -------------------------------------------------------------------------------
+# Evolutionary optimization: Parallel evaluation
+# -------------------------------------------------------------------------------
+def evaluator(candidates, args):
+    import os
+    global ngen
+    ngen += 1
+    total_jobs = 0
+    # read params or set defaults
+    sleepInterval = args.get('sleepInterval', 1)
+    
+    # paths to required scripts
+    script = '../' + args.get('script', 'init.py')
+    cfgSavePath = '../' + args.get('cfgSavePath')
+    netParamsSavePath = '../' + args.get('netParamsSavePath')
+    simDataFolder = args.get('simDataFolder') + '/gen_' + str(ngen)
+    
+    # mpi command setup
+    nodes = args.get('nodes', 1)
+    paramNames = args.get('paramNames')
+    coresPerNode = args.get('coresPerNode', 1)
+    mpiCommand = args.get('mpiCommand', 'ibrun')
+    numproc = nodes*coresPerNode
+    
+    # slurm setup
+    custom = args.get('custom', '')
+    folder = args.get('folder', '.')
+    email = args.get('email', 'a@b.c')
+    walltime = args.get('walltime', '00:30:00')
+    reservation = args.get('reservation', None)
+    allocation = args.get('allocation', 'csd403') # NSG account
+    if reservation:
+        res = '#SBATCH --res=%s'%(reservation)
+    else:  
+        res = ''
+        
+    # fitness function
+    fitness_expression = args.get('fitness')
+    default_fitness = args.get('default_fitness')
+    
+    # create a folder for this generation
+    if not os.path.exists(simDataFolder):
+        try:
+            os.mkdir(simDataFolder)
+        except OSError:
+            print ' Could not create %s' %(simDataFolder)
+    
+    # create a job for each candidate
+    for candidate_index, canditate in enumerate(candidates):
+        # required for slurm
+        sleep(sleepInterval)
+        jobName = "gen_" + str(ngen) + "_cand_" + str(candidate_index)
+        simDataPath = simDataFolder + '/' + jobName
+        # script to run with mpi
+        command = '%s -np %d nrniv -python -mpi %s simConfig=%s netParams=%s filename=%s ' % (mpiCommand, numproc, script, cfgSavePath, netParamsSavePath, simDataPath)
```

Comment:
note that filename=%s is not valid and not required anymore; the same applies for the next 3 lines which add the parameter values via command line -- that was the old method, but now we are using the netpyne method, which consists of modifying the parameters in cfg.py. So the job submission should be identical to the 'grid' method, except that the parameters are generated by the evol algorithm instead of the grid search method.

Reply:
this is the line where the 'grid' method modifies the parameters via the cfg object: https://github.com/Neurosim-lab/netpyne/blob/development/netpyne/batch.py#L186
the evol algorithm should do the same